src/main/kotlin/it/pagopa/wallet/client/WalletQueueClient.kt

```diff
@@ -23,8 +25,12 @@ class WalletQueueClient(
         tracingInfo: QueueTracingInfo
     ): Mono<Response<SendMessageResult>> {
         val queueEvent = QueueEvent(event, tracingInfo)
-        return BinaryData.fromObjectAsync(queueEvent, jsonSerializer).flatMap {
-            cdcQueueClient.sendMessageWithResponse(it, delay, ttl)
-        }
+        return BinaryData.fromObjectAsync(queueEvent, jsonSerializer)
+            .flatMap { cdcQueueClient.sendMessageWithResponse(it, delay, ttl) }
+            .retryWhen(
+                Retry.backoff(3, Duration.of(1, ChronoUnit.SECONDS)).doBeforeRetry { signal ->
+                    println("Retrying due to: ${signal.failure().message}")
+                }
+            )
```

Comment:
isnt it more compact and readable to have the retry mechanism in place here on the sendMessage function?

Reply:
We could move the retry logic inside the WalletPaymentCDCEventDispatcherService class, something like this:
```
fun dispatchEvent(event: BsonDocument?): Mono<Response<SendMessageResult>> =
        if (event != null) {
            Mono.defer { onWalletEvent(event) }
                .retryWhen(
                    Retry.fixedDelay(
                            retrySendPolicyConfig.maxAttempts,
                            Duration.ofMillis(retrySendPolicyConfig.intervalInMs)
                        )
                        .filter { t -> t is Exception }
                ).repeat(response != success)
        } else {
            Mono.empty()
        }
```

Reply:
yes, this is exactly what i was doing 
:)

Reply:
done here e9418a32ae033888a9edb0b2ec087bac0a56001d

Reply:
Let's have business logic inside the service and not inside the queueClient, consider moving the retry logic inside WalletPaymentCDCEventDispatcherService from WalletQueueClient. Furthermore, we are only covering retries in case an exception occurs, whe should add retries also based on Response<SendMessageResult> evaluation but we could take care of this in a dedicated PR.
Please add test cases to check if the retry mechanism behaves as expected.

Reply:
what is the point of .repeat(response != success)?
didn't we already filter exceptions using          
               .filter { t -> t is Exception }

Reply:
Sending a message could result in a failure wrapped in a response code not equal to 2xx instead of an exception thrown. We can take care of this case in a dedicated PR later on